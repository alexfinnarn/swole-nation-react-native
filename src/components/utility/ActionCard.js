import React from 'react';
import {Text, View} from 'react-native';
import {home} from '../Styles';
import ActionButton from './ActionButton';

export default function ActionCard({children, actionComponent}) {
  return (
    <View style={home.sectionContainer}>
      <View style={home.sectionLeft}>{children}</View>
      {actionComponent}
    </View>
  );
}
