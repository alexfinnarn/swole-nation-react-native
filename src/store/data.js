import addFive from '../transformers/addFive'

const data = {
  workouts: {
    DQkECwYLCQQ: {
      name: "Stronglifts A",
      key: "DQkECwYLCQQ",
      description: "Squats, Bench Press, Barbell Row",
      exercises: [
        "Squats Warmup",
        "Squats",
        "Bench Press Warmup",
        "Bench Press",
        "Barbell Row Warmup",
        "Barbell Row"
      ]
    },
    DAwJBQEMAA0: {
      name: "Stronglifts B",
      key: "DAwJBQEMAA0",
      description: "Squats, Overhead Press, Deadlifts",
      exercises: [
        "Squats Warmup",
        "Squats",
        "Overhead Press Warmup",
        "Overhead Press",
        "Deadlifts Warmup",
        "Deadlifts"
      ]
    }
  },
  exercises: {
    BwEBDwoFBgQ: {
      name: "Squats Warmup",
      key: "BwEBDwoFBgQ",
      image: require("../../assets/squats.gif"),
      instructions: "Do with a barbell",
      sets: [
        {
          "reps": 5,
          "weight": 45.0
        },
        {
          "reps": 5,
          "weight": 45.0
        },
        {
          "reps": 5,
          "weight": 95.0
        },
        {
          "reps": 5,
          "weight": 135.0
        },
        {
          "reps": 3,
          "weight": 185.0
        },
        {
          "reps": 2,
          "weight": 205.0
        }
      ]
    },
    Bw0ECwAKBws: {
      name: "Squats",
      key: "Bw0ECwAKBws",
      image: require("../../assets/squats.gif"),
      instructions: "Do with a barbell",
      sets: [
        {
          "reps": 5,
          "weight": 230.0
        },
        {
          "reps": 5,
          "weight": 230.0
        },
        {
          "reps": 5,
          "weight": 230.0
        },
        {
          "reps": 5,
          "weight": 230.0
        },
        {
          "reps": 5,
          "weight": 230.0
        }
      ]
    },
    BwwBDAIJDgw: {
      name: "Bench Press Warmup",
      key: "BwwBDAIJDgw",
      image: require("../../assets/bench-press.gif"),
      instructions: "Do with a barbell",
      sets: [
        {
          "reps": 5,
          "weight": 45.0
        },
        {
          "reps": 5,
          "weight": 45.0
        },
        {
          "reps": 5,
          "weight": 95.0
        },
        {
          "reps": 3,
          "weight": 135.0
        }
      ]
    },
    DgcBCA8BCQo: {
      name: "Bench Press",
      key: "DgcBCA8BCQo",
      image: require("../../assets/bench-press.gif"),
      instructions: "Do with a barbell",
      sets: [
        {
          "reps": 5,
          "weight": 165.0
        },
        {
          "reps": 5,
          "weight": 165.0
        },
        {
          "reps": 5,
          "weight": 165.0
        },
        {
          "reps": 5,
          "weight": 165.0
        },
        {
          "reps": 5,
          "weight": 165.0
        }
      ]
    },
    DwcACw4PCgI: {
      name: "Shoulder Press Warmup",
      key: "DwcACw4PCgI",
      image: require("../../assets/shoulder-press.gif"),
      instructions: "Do with a barbell",
      sets: [
        {
          "reps": 5,
          "weight": 45.0
        },
        {
          "reps": 5,
          "weight": 45.0
        },
        {
          "reps": 3,
          "weight": 75.0
        }
      ]
    },
    CAMBAwsFBwQ: {
      name: "Shoulder Press",
      key: "CAMBAwsFBwQ",
      image: require("../../assets/shoulder-press.gif"),
      instructions: "Do with a barbell",
      sets: [
        {
          "reps": 5,
          "weight": 110.0
        },
        {
          "reps": 5,
          "weight": 110.0
        },
        {
          "reps": 5,
          "weight": 110.0
        },
        {
          "reps": 5,
          "weight": 110.0
        },
        {
          "reps": 5,
          "weight": 110.0
        }
      ]
    },
    BwYGAw0NDwo: {
      name: "Barbell Row Warmup",
      key: "BwYGAw0NDwo",
      image: require("../../assets/barbell-row.gif"),
      instructions: "Do with a barbell",
      sets: [
        {
          "reps": 5,
          "weight": 95.0
        }
      ]
    },
    BwQKAwYKAgA: {
      name: "Barbell Row",
      key: "BwQKAwYKAgA",
      image: require("../../assets/barbell-row.gif"),
      instructions: "Do with a barbell",
      sets: [
        {
          "reps": 5,
          "weight": 115.0
        },
        {
          "reps": 5,
          "weight": 115.0
        }
      ]
    },
    AQcNAQkBCwo: {
      name: "Deadlifts Warmup",
      key: "AQcNAQkBCwo",
      image: require("../../assets/deadlift.gif"),
      instructions: "Do with a barbell",
      sets: [
        {
          "reps": 5,
          "weight": 135.0
        },
        {
          "reps": 5,
          "weight": 185.0
        },
        {
          "reps": 5,
          "weight": 205.0
        }
      ]
    },
    DAYCDAQKAw4: {
      name: "Deadlifts",
      key: "DAYCDAQKAw4",
      image: require("../../assets/deadlift.gif"),
      instructions: "Do with a barbell",
      sets: [
        {
          "reps": 5,
          "weight": 225.0
        }
      ]
    }
  },
  sessions: {
    BwgFDQ8CCg8: {
      name: "new ...date",
      key: "BwgFDQ8CCg8",
      workoutKeys: [],
      exercises: []
    }
  },
  transformers: {
    none: {
      key: 'none',
      label: "N/A",
      callback: null
    },
    addFive: {
      key: 'addFive',
      label: "Exercises (+5 lbs)",
      callback: addFive
    }
  },
  activeWorkoutKey: "",
  activeExerciseKey: "",
  activeTransformerKey: "none",
};

export default data;
