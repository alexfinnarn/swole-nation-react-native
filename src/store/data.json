{
  "workouts": [
    {
      "name": "Stronglifts A",
      "key": "DQkECwYLCQQ",
      "description": "Squats, Bench Press, Barbell Row",
      "exercises": [
        "Squats Warmup"
      ]
    }
  ],
  "exercises": [
    {
      "name": "Squats Warmup",
      "key": "BwEBDwoFBgQ",
      "image": "../../assets/squats.gif",
      "instructions": "Do with a barbell",
      "sets": [
        {
          "reps": 5,
          "weight": 45.0
        },
        {
          "reps": 5,
          "weight": 45.0
        },
        {
          "reps": 5,
          "weight": 95.0
        },
        {
          "reps": 5,
          "weight": 135.0
        },
        {
          "reps": 3,
          "weight": 185.0
        },
        {
          "reps": 2,
          "weight": 205.0
        }
      ]
    }
  ],
  "sessions": [
    {
      "name": "new ...date",
      "key": "BwgFDQ8CCg8",
      "workoutKeys": [],
      "exercises": []
    }
  ],
  "tempExerciseList": [],
  "theThing": "",
  "activeWorkoutIndex": 0,
  "activeSessionIndex": 0,
  "activeExerciseIndex": 0
}
