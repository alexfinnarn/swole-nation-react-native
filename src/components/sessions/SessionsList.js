import React from 'react';
import {FlatList, Text, View} from "react-native";
import {home, styles} from '../Styles';
import ActionButton from "../utility/ActionButton";
import ActionCard from "../utility/ActionCard";
import SessionTeaser from "./SessionTeaser";

export default function SessionsList({ sessions, navigation, handle, thing }) {
  return (
    <View style={styles.container} testID="sessions-list-root">
      <View style={{flex:9}}>
        <FlatList
          extraData={thing}
          data={sessions}
          ListEmptyComponent={(<Text>No sessions to list.</Text>)}
          renderItem={(item) => <SessionsListItem session={item}/>}
        />
      </View>
      {/*<ActionButton text="Add session" action={() => navigation.navigate('Session')} />*/}
    </View>
  );

  function SessionsListItem({session}) {
    return (
      <ActionCard actionComponent={
        <>
          <ActionButton styles={{marginRight: 2}} text="Edit" action={() => {
            handle.setActiveSessionKey(session.item.key);
            navigation.navigate('Session', {sessionId: session.item.key, title: session.item.name})
          }}/>
          <ActionButton text="X" action={() => {
            handle.deleteSession(session.item.key);
            // navigation.navigate('Home');
          }}/>
        </>
      }>
        <SessionTeaser session={session.item}/>
      </ActionCard>
    );
  }
}



