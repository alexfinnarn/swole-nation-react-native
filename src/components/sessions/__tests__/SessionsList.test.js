import React from 'react';
import SessionsList from '../SessionsList';
import { render, fireEvent } from 'react-native-testing-library';
import data from '../../../store/data';
import ActionCard from "../../utility/ActionCard";
import shortId from "shortid";

let renderer, instance ={};
let sessions = Object.keys(data.sessions).map(key => data.sessions[key]);
const handle = {
  setActiveSessionKey: jest.fn(),
  deleteSession: jest.fn(),
};
const navigation = {
  navigate: jest.fn()
};

beforeEach(() => {
  renderer = render(<SessionsList sessions={sessions} navigation={navigation} handle={handle} thing={shortId.generate()}/>);
  instance = renderer.getByTestId('sessions-list-root');
}, 0);

describe('<SessionsList />', () => {
  // @todo Foiled by the extraData param that is always unique. I shouldn't need that data to re-render so I should remove it.
  // it('Renders correctly', () => {
  //   expect(renderer.toJSON()).toMatchSnapshot();
  // });

  it('Should render one session in the list initially and two when session added', () => {
    let actionCards = instance.findAllByType(ActionCard);
    expect(actionCards.length).toBe(1);
    expect(renderer.getAllByText('Edit').length).toBe(1);

    expect(renderer.queryByText('Workout C')).toBeNull();

    sessions.push({key: 'DWYVDAQWAw4', name: 'Workout C', description: 'blah', exercises: []});
    renderer.update(<SessionsList sessions={sessions} navigation={navigation} handle={handle} thing={shortId.generate()} />);
    instance = renderer.getByTestId('sessions-list-root');

    actionCards = instance.findAllByType(ActionCard);
    expect(actionCards.length).toBe(2);
    expect(renderer.getAllByText('Edit').length).toBe(2);

    expect(renderer.queryByText('Workout C')).not.toBeNull();
  });

  it('Should navigate to the Session screen when edit button pressed', () => {
    // Expect navigate to first session.
    fireEvent(renderer.getAllByText('Edit')[0], 'press');
    expect(navigation.navigate).toHaveBeenCalledWith('Session', {sessionId: "BwgFDQ8CCg8"});
    expect(handle.setActiveSessionKey).toHaveBeenCalledTimes(1);
    expect(handle.setActiveSessionKey).toHaveBeenCalledWith('BwgFDQ8CCg8');
  });

  it('Should delete a session and refresh the flat list', () => {
    let actionCards = instance.findAllByType(ActionCard);
    expect(actionCards.length).toBe(2);
    expect(renderer.queryByText('No sessions to list.')).toBeNull();

    fireEvent(renderer.getAllByText('X')[0], 'press');
    sessions = [];
    renderer.update(<SessionsList sessions={sessions} navigation={navigation} handle={handle} thing={shortId.generate()} />);

    actionCards = instance.findAllByType(ActionCard);
    expect(actionCards.length).toBe(0);
    expect(renderer.queryByText('No sessions to list.')).not.toBeNull();
    expect(handle.deleteSession).toHaveBeenCalledTimes(1);
    expect(handle.deleteSession).toHaveBeenCalledWith('BwgFDQ8CCg8');
  });
});
