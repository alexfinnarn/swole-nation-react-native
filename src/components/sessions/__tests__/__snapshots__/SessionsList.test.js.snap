// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SessionsList /> Renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#F5FCFF",
      "flex": 1,
      "flexDirection": "column",
      "paddingTop": 5,
    }
  }
  testID="sessions-list-root"
>
  <View
    style={
      Object {
        "flex": 9,
      }
    }
  >
    <RCTScrollView
      data={
        Array [
          Object {
            "exercises": Array [],
            "key": "BwgFDQ8CCg8",
            "name": "new ...date",
            "workoutKeys": Array [],
          },
        ]
      }
      disableVirtualization={false}
      extraData="tFI_8PPX"
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      numColumns={1}
      onContentSizeChange={[Function]}
      onEndReachedThreshold={2}
      onLayout={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      updateCellsBatchingPeriod={50}
      viewabilityConfigCallbackPairs={Array []}
      windowSize={21}
    >
      <View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              Object {
                "backgroundColor": "#ffffff",
                "elevation": 5,
                "flex": 1,
                "flexDirection": "row",
                "marginBottom": 4,
                "marginTop": 4,
                "shadowColor": "#000",
                "shadowOffset": Object {
                  "height": 2,
                  "width": 0,
                },
                "shadowOpacity": 0.25,
                "shadowRadius": 3.84,
              }
            }
          >
            <View
              style={
                Object {
                  "flex": 4,
                  "flexDirection": "column",
                  "padding": 10,
                }
              }
            >
              <Text
                style={
                  Object {
                    "flex": 1,
                    "fontSize": 18,
                  }
                }
              >
                new ...date
              </Text>
              <Text
                style={
                  Object {
                    "flex": 4,
                    "marginTop": 15,
                  }
                }
              >
                Other content
              </Text>
            </View>
            <View
              style={
                Array [
                  Object {
                    "flex": 1,
                    "flexDirection": "column",
                  },
                  undefined,
                ]
              }
            >
              <View
                accessible={true}
                isTVSelectable={true}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignSelf": "stretch",
                    "backgroundColor": "#21897E",
                    "flex": 1,
                    "opacity": 1,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "flex": 1,
                      "justifyContent": "center",
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "alignSelf": "center",
                        "color": "#ffffff",
                        "fontSize": 20,
                      }
                    }
                  >
                    Edit
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</View>
`;
